tools/src/main/java/org/apache/kafka/tools/StreamsResetter.java

```diff
@@ -584,8 +584,8 @@ public StreamsResetterOptions(String[] args) {
                 .ofType(String.class)
                 .withValuesSeparatedBy(',')
                 .describedAs("list");
-            intermediateTopicsOption = parser.accepts("intermediate-topics", "Comma-separated list of intermediate user topics (topics that are input and output topics, "
-                    + "e.g., used in the deprecated through() method). For these topics, the tool will skip to the end.")
+            intermediateTopicsOption = parser.accepts("intermediate-topics", "Comma-separated list of intermediate user topics (topics that are input and output topics). "
```

Comment:
I think, we can remove this options from the tool entirely... (Or would we need to keep it as we did not really officially deprecate it... 🤔 -- curious to hear what @ableegoldman @cadonna @lucasbru @bbejeck think...)

Reply:
That's the question -- if one uses an older version of KS, would they also use an older version of the tool?

I am open both ways, but yes, we might want to remove it in 5.0 if we keep it for now.

Reply:
I think we should remove it in 5.0, because @lucasbru makes a good point about migration and resetting a deprecated application.

Reply:
I tend to agree -- let's make sure to "officially" deprecate it in 4.0 though, ie we should log a warning if this option is used

Reply:
got, it I will create a commit that will add a warning log if this option is used.

Reply:
If we want to deprecate is, should we do a small KIP about it?

Reply:
I don't really think we need to do a KIP, since through is now removed. But I am not opposed to doing a small KIP either.

Reply:
@lucasbru @ableegoldman What do you think? Do we need a KIP?

Reply:
I'm personally fine with skipping a KIP...though it's probably correct to do one. I'd say just write up a short and quick KIP and immediately start the voting. This way there's at least a record of when/why was this was deprecated and people are more likely to notice (especially since there won't be a deprecation warning and all we can do is log a warning and hope people see it)

Reply:
Got it I will write a short KIP

Reply:
I have requested permission on the WIKI to make a KIP, but still have not received it. This may take a while, as I cannot make a KIP without the wiki admins giving me permissions .

Reply:
I don't see any request which I could map to you. Did you use the new self-service portal? What user name did you pick (also, did you pick Kafka as project)?

Reply:
Ah. Never mind. Your email was not going through automatically (seems, you did not sign up correctly for the dev mailing list -- should be fixed now, too).

Reply:
No, I was one of the last people to be issued an account via the old method

my user name is : ardada2468

I think that I have selected Kafka as the project.

Reply:
[KIP-1087 ](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1087%3A+Removing+intermediateTopicsOption+from+StreamsResetter): I have just created it, this is my first time

Reply:
Thanks. Please start a DISCUSS thread on the dev mailing list about it.

Reply:
just sent

Reply:
I have sent the VOTE and DISCUSS thread

Reply:
I think we should update the description for this option, adding `[deprecated]` and maybe update to "no-op" or similar?

Reply:
@ardada2468 Seems you did not address this comment yet?